src/test/scala/com/microsoft/hyperspace/index/DeltaLakeIntegrationTest.scala

```diff
@@ -68,7 +68,9 @@ class DeltaLakeIntegrationTest extends QueryTest with HyperspaceSuite {
       dfFromSample.write.format("delta").save(dataPath)
 
       val deltaDf = spark.read.format("delta").load(dataPath)
-      hyperspace.createIndex(deltaDf, IndexConfig("deltaIndex", Seq("clicks"), Seq("Query")))
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
```

Comment:
why is this required now?

Reply:
Since I added the condition above:
https://github.com/microsoft/hyperspace/pull/389/files/b938556191c3dcceab0ec784fa26defc5b02804b#diff-ccc638dd1aab8f054c6d13cdfafeea9c1634646e90dd999b491e4e43a18d9efbR188

It's possible to allow append-only hybrid scan if possible, but the condition in `closestIndex` will become more complicated. So just limit the case for now.

Reply:
Hmm, so before this PR, append-only hybrid scan was supported? I am curious which change in this PR triggered this new requirement.

Reply:
Might be supported. The index was not applied because of the conditions in getHybridScanCandidate if the version has deleted files (compared to the given relation) - before this PR. 

I found that I was writing a test for directory check change.